lib/iris/_lazy_data.py

```diff
@@ -135,3 +135,64 @@ def multidim_lazy_stack(stack):
         result = da.stack([multidim_lazy_stack(subarray)
                            for subarray in stack])
     return result
+
+
+def convert_nans_array(array, nans=None, result_dtype=None):
+    """
+    Convert a :class:`~numpy.ndarray` that may contain one or more NaN values
+    to either a :class:`~numpy.ma.core.MaskedArray` or a
+    :class:`~numpy.ndarray` with the NaN values filled.
+
+    Args:
+
+    * array:
+        The :class:`~numpy.ndarray` to be converted.
+
+    Kwargs:
+
+    * nans:
+        If `nans` is None, then raise an exception if the `array` contains
+        any NaN values (default).
```

Comment:
It's the default in the function formal parameter list. 

Also, it's a very useful value to default to because it's gonna raise an exception if it detects that there are NaNs in your data. We're forcing people to care about this. So for example, this is good behaviour to adopt as it will force the issue in cases when streamed saving is being attempted with NaN data, but the `fill_value` isn't set i.e. None. See [here](https://github.com/SciTools/iris/pull/2411/files#diff-e055d3568b9b3f9cfe397cec7e016700R1955) in particular.